File: WordGames/Cat5Letters/CAT5long.py
# Program to create a list for all of the words containing CAT and are exactly 5 letters long?


# importing the word list
sow_pods = open("/home/evergreen/UnderDogsAssignments/sowpods")

# Go through each word in the file and add it to the list, words
words = []
cat = []
cat5 = []
x = 0

for line in sow_pods:
    words.append(line.strip())

for w in words:
  if ("CAT" in w):
      cat.append(w)

for x in cat:
    if len(x) == 5:
        cat5.append(x)
        #x = x + 1
    
print('All of the words that have "CAT" and are 5 letters long are:' + str(cat5))

File: WordGames/Cat5Letters/cat5Long.js
// Program to find all the words containing cat and are 5 letters long
// Not currently working
// Importing the fs module
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let words = readFileLines('/home/evergreen/UnderDogsAssignments/sowpods');
cat = []
cat5 = []

for(let i = 0; i < words.length; i++) { 
    if (words[i].includes("CAT")) {
        cat.push(words[i]);
    }

for(let j = 0; j < cat.length; j++) { 
    let long = cat[j].length;
    if (long = 5) {
        cat5.push(cat[j]);
        }
        }

}
console.log(cat5)

//console.log("All the words with 2 consecutive Us are " + cat)}
